### PR TITLE
Prevent elder guardians from giving mining fatigue within 100 blocks of spawn

### PR DESCRIPTION
**Pull request by MoWobbler**

### CODE REVIEW

**Comment by C4K3** [comment]
Thanks. Let's make it a configuration option to enable so we can disable it on the build server. And also let's make it only activate in the overworld. (Maybe the configuration option can be a list of worlds it's enabled on?)

**Comment by MoWobbler** [comment]
I was testing this and it seemed to be working, but I didn't test if it works for multiple worlds. The disableElderGuardians thing in the config is a list of worlds

File: src/main/java/net/simpvp/Misc/PotionEffectListener.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.bukkit.potion.PotionEffectType;
+
+//This file disables elder guardians from giving mining fatigue within 100 blocks of spawn
+public class PotionEffectListener implements Listener {
+	
+	@EventHandler
+	public void potionEffectEvent(EntityPotionEffectEvent event) {
+		
+		//Check if world name is in the config
+		if (!Misc.instance.getConfig().getList("disableElderGuardians").contains(event.getEntity().getWorld().getName())) {
```
Comment (C4K3) [inline src/main/java/net/simpvp/Misc/PotionEffectListener.java]: I think getting the list is a bit more expensive than keeping it in a local variable. Could you load the config list into a private HashSet in the class when initializing the plugin?

File: src/main/java/net/simpvp/Misc/PotionEffectListener.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+			return;
+		}
+		
+		//If entity is less than 100 blocks from spawn
+		if (event.getEntity().getLocation().distance(event.getEntity().getWorld().getSpawnLocation()) <= 100) {
```
Comment (C4K3) [inline src/main/java/net/simpvp/Misc/PotionEffectListener.java]: Use `.distanceSquared() <= 100*100`.

**Comment by MoWobbler** [comment]
Would something like this work?

**Comment by C4K3** [comment]
Thanks!

**Pull request merged**